lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -36,11 +47,15 @@ def __init__(self, config):
         self.dfu = DataFileUtil(config.callbackURL)
         self.aUtil = AssemblyUtil(config.callbackURL)
         self.ws = Workspace(config.workspaceURL)
+        self.re_api_url = config.re_api_url
+        yml_text = open('/kb/module/kbase.yml').read()
+        self.version = re.search("module-version:\n\W+(.+)\n", yml_text).group(1)
+        self.reset_attributes()
+
+    def reset_attributes(self):
```

Comment:
using the instance variables of a class over and over for what are effectively different instances of that class is asking for people to not understand what's going on in the future and introduce bugs.

Instead, refactor this class to be able to handle more than one "genome" instance at the same time. What I would probably do is make a new class (maybe in this module) that contains the per "genome" instance variables and refactor the code to make a new class every time a new genome is processed. 

I'm not sure if we're parallelizing via threading or coroutines but if we are / did then the threads would all be clobbering each other if we only had the one class instance being reused for all "genomes". 

This is a pretty large refactor so I'm going to leave the review here for now assuming that the code is going to change quite a bit

Reply:
I made a new class named `Genome` in the same file. Not quite sure what you mean `(maybe in this module)`

Reply:
Update: I move `Genome` class to a separate file cause its setters and getters are getting extremely long.

Reply:
In python `module` == `file`

Reply:
> Update: I moved Genome class to a separate file cause its setters and getters are getting extremely long.

In python in general you don't do getters and setters, you just access the instance vars directly. The class author can always add the `__getattr__` and `__setattr__` methods if they need to. And since this should be a class for internal use only you probably don't need to be all that strict about it, esp if it's in the same module and marked as private

Reply:
I was thinking make all attributes private and then add @property @<attribute>.setter so that we can still access those private vars like public. For instance, self.code_table = 12.

But if you think we don't need to be strict about it. That's also cool. Indeed, less work :)

Reply:
If it's in the same module & private, and is therefore just an invisible (to the module API) helper class and almost analogous to a c struct, I don't think you need to put a ton of effort into it

Reply:
👍

Reply:
:+1: